app/controllers/concerns/foreman_remote_execution/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,36 @@
+module ForemanRemoteExecution
+  module HostsControllerExtensions
+    extend ActiveSupport::Concern
+    include Foreman::Renderer
+
+    included do
+      alias_method_chain(:action_permission, :remote_execution)
+    end
+
+    def reprovision
+      find_resource
+      script_template = @host.provisioning_template(:kind => 'script')
+      if script_template.nil?
+        process_error :redirect => :back, :error_msg => _("No script provisioning template available")
+        return
+      end
+      @host.setBuild
+      script = unattended_render(script_template.template, @template_name)
+      composer = JobInvocationComposer.for_feature(:reprovision, @host, :script => script)
+      composer.save!
+      composer.trigger
```

Comment:
We do this 3-step dance in a lot of places, we don't have to change it now, but any reason not to just save and trigger in one step for this and other controllers? Something like 

```ruby
JobInvocationComposer.trigger_for_feature!(:reprovision, @host, :script => script)
```